http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala

```diff
@@ -125,6 +125,27 @@ class HttpSamDAO[F[_]](httpClient: Client[F],
       )(onError)
   }
 
+  /** For every role the user is granted on a Sam resource, list it and the resource ID. */
+  override def listResourceIdsWithRole[R <: SamResourceId](
```

Comment:
Does this method essentially duplicate `getResourcePolicies` -- except that we're no longer mis-construing Sam roles as policy names?

I'm all for fixing the naming -- but to avoid confusion could we remove or at least `@deprecate` the other method and related models (like `SamPolicyName`, `listResourceResponseDecoder`, `ListResourceResponse`, etc).

Reply:
Correct. I will @deprecate!